client/stripe-utils/__tests__/utils.test.js

```diff
@@ -39,4 +39,208 @@ describe( 'utils', () => {
 			expect( result ).toBe( expectedFontSize );
 		} );
 	} );
+
+	describe( 'getDefaultValues', () => {
+		const globalValues = global.wc_stripe_upe_params;
+		let mockGetElementById;
+
+		beforeEach( () => {
+			global.wc_stripe_upe_params = {
+				isOCEnabled: false,
+			};
+
+			// Mock document.getElementById for fallback behavior
+			mockGetElementById = jest.fn();
+			document.getElementById = mockGetElementById;
+		} );
+
+		afterEach( () => {
+			global.wc_stripe_upe_params = globalValues;
+			jest.restoreAllMocks();
+		} );
+
+		describe( 'when isOrderPay, isChangingPayment, or isAddPaymentMethod is true', () => {
+			it( 'should return correctly formatted billing data from customerBillingData', () => {
+				global.wc_stripe_upe_params = {
+					isOrderPay: true,
+					customerBillingData: {
+						name: 'John Doe',
+						email: 'john@example.com',
+						phone: '+1234567890',
+						address: {
+							country: 'us', // lowercase, should be uppercased
+							line1: '123 Main St',
+							line2: 'Apt 4B',
+							city: 'New York',
+							state: 'NY',
+							postal_code: '10001',
+						},
+					},
+				};
+
+				const result = getDefaultValues();
+
+				expect( result ).toEqual( {
+					defaultValues: {
+						billingDetails: {
+							name: 'John Doe',
+							email: 'john@example.com',
+							phone: '+1234567890',
+							address: {
+								country: 'US', // Should be uppercase
+								line1: '123 Main St',
+								line2: 'Apt 4B',
+								city: 'New York',
+								state: 'NY',
+								postal_code: '10001',
+							},
+						},
+					},
+				} );
+			} );
+
+			it( 'should filter out empty address fields and trim whitespace', () => {
+				global.wc_stripe_upe_params = {
+					isOrderPay: true,
+					customerBillingData: {
+						name: '  John Doe  ',
+						email: '  john@example.com  ',
+						phone: '  +1234567890  ',
+						address: {
+							country: '  us  ',
+							line1: '  123 Main St  ',
+							line2: '', // empty, should be filtered out
+							city: '  New York  ',
+							state: '', // empty, should be filtered out
```

Comment:
Nit: it feels like it would be worth making one of these fields whitespace only to check that we handle that correctly.
```suggestion
							state: '    ', // only whitespace, should be filtered out
```